executor/inspection_result.go

```diff
@@ -663,3 +665,139 @@ func (thresholdCheckInspection) inspectThreshold2(ctx context.Context, sctx sess
 	}
 	return results
 }
+
+type thresholdCheckRule interface {
+	genSQL(timeRange plannercore.QueryTimeRange) string
+	genResult(sql string, row chunk.Row) inspectionResult
+	getItem() string
+}
+
+type compareStoreStatus struct {
+	item      string
+	tp        string
+	threshold float64
+}
+
+func (c compareStoreStatus) genSQL(timeRange plannercore.QueryTimeRange) string {
+	condition := fmt.Sprintf(`where t1.time>='%[1]s' and t1.time<='%[2]s' and
+		 t2.time>='%[1]s' and t2.time<='%[2]s'`, timeRange.From.Format(plannercore.MetricTableTimeFormat),
+		timeRange.To.Format(plannercore.MetricTableTimeFormat))
+	return fmt.Sprintf(`select t1.address,t1.value,t2.address,t2.value,
+				(t1.value-t2.value)/greatest(t1.value,t2.value) as ratio
+				from metrics_schema.pd_scheduler_store_status t1 join metrics_schema.pd_scheduler_store_status t2
+				%s and t1.type='%s' and
+				t1.type=t2.type and t1.address != t2.address and
+				(t1.value-t2.value)/t1.value>%v;`, condition, c.tp, c.threshold)
+}
+
+func (c compareStoreStatus) genResult(_ string, row chunk.Row) inspectionResult {
+	addr1 := row.GetString(0)
+	value1 := row.GetFloat64(1)
+	addr2 := row.GetString(2)
+	value2 := row.GetFloat64(3)
+	ratio := row.GetFloat64(4)
+	expected := fmt.Sprintf("The difference between %v and %v should less than %v%%, actual is %v%%", addr1, addr2, c.threshold*100, ratio*100)
```

Comment:
I think the `expected` is too verbose.
We can rename the `item` to `region-score-balance` and set the `expected` to the threshold,  the `ratio` information should be put in `detail`.